seqr/views/apis/anvil_workspace_api.py

```diff
@@ -449,4 +458,25 @@ def _make_airflow_api_request(endpoint, method='GET', timeout=90, **kwargs):
     return resp.json()
 
 
+def _get_common_prefix(files):
+    for i in range(len(files[0])):
+        if any([file[i] != files[0][i] if i < len(file) else True for file in files]):
+            return files[0][:i]
+    return ''
+
+
+def _merge_sharded_vcf(vcf_files):
+    files_by_path = defaultdict(list)
+
+    for vcf_file in vcf_files:
+        subfolder_path, file = vcf_file.rsplit('/', 1)
+        files_by_path[subfolder_path].append(file)
+
+    # discover the sharded VCF files in each folder, replace the sharded VCF files with a single path with '*'
+    for subfolder_path, files in files_by_path.items():
+        prefix = _get_common_prefix(files)
+        postfix = files[0].replace(prefix, '').lstrip('_0123456789')
```

Comment:
the opposite of prefix is suffix, not postfix. Also, this works but I think its a bit more explicit using regex:
```
if len(files) < 2:
    continue
prefix = os.path.commonprefix(files)
suffix = re.fullmatch(f'{prefix}\d+(?P<suffix>\D+)', files[0]).groupdict()['suffix']
if all(re.fullmatch(f'{prefix}\d+{suffix}', file) for file in files]):
    ...
```